crypto/src/vrf/mod.rs

```diff
@@ -151,6 +163,101 @@ impl VRFPublicKey {
     }
 }
 
+/// Given a tree of keys that are derived from a master key using BIP32 rules, this struct represents
+/// the private key at one of the nodes of this tree.
+#[derive(Debug, PartialEq, Eq, Clone, Encode, Decode)]
+pub struct ExtendedVRFPrivateKey {
+    /// The absolute derivation path that was used to derive this key
+    derivation_path: DerivationPath,
+    /// The chain code is used in BIP32 in conjunction with the private key to allow derivation
+    /// of child keys
+    chain_code: ChainCode,
+    /// The private key to be used to derive child keys of this node using BIP32 rules
+    private_key: VRFPrivateKey,
+}
+
+impl ExtendedVRFPrivateKey {
+    pub fn new_master(
+        seed: &[u8],
+        keykind: VRFKeyKind,
+    ) -> Result<ExtendedVRFPrivateKey, DerivationError> {
+        // Create a new mac with the appropriate BIP39 constant
+        let mut mac = new_hmac_sha_512(b"Bitcoin seed");
+
+        mac.update(seed);
+
+        let (private_key, chain_code) = to_key_and_chain_code(mac, keykind)?;
```

Comment:
I know the point was to separate this from the normal key derivation. However, as a result there is some duplication, like this bit, the `new_hmac_sha_512` helper, etc. Is it possible/sensible to factor out the common parts into a shared module?

Reply:
moved the 2 functions , `new_hmac_sha_512` and `to_key_and_chain_code` to the util module